instrumentation/java.logging-jdk8/src/main/java/java/util/logging/Logger_Instrumentation.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *  * Copyright 2022 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package java.util.logging;
+
+import com.newrelic.api.agent.NewRelic;
+import com.newrelic.api.agent.weaver.Weave;
+import com.newrelic.api.agent.weaver.Weaver;
+
+import static com.nr.instrumentation.jul.AgentUtil.isApplicationLoggingEnabled;
+import static com.nr.instrumentation.jul.AgentUtil.isApplicationLoggingMetricsEnabled;
+
+@Weave(originalName = "java.util.logging.Logger")
+public class Logger_Instrumentation {
+
+    public Filter getFilter() {
+        return Weaver.callOriginal();
+    }
+
+    public boolean isLoggable(Level level) {
+        return Boolean.TRUE.equals(Weaver.callOriginal());
+    }
+
+    public void log(LogRecord record) {
+        // Do nothing if application_logging.enabled: false
+        if (isApplicationLoggingEnabled()) {
+            if (isApplicationLoggingMetricsEnabled()) {
+                if (isLoggable(record.getLevel()) && getFilter() == null || getFilter().isLoggable(record)) {
```

Comment:
Can record ever be null at this point?

Reply:
They don't seem to be doing null checks on `record` in the library method that we're weaving this into so I think we're probably fine using it as is.

```
    /**
     * Log a LogRecord.
     * <p>
     * All the other logging methods in this class call through
     * this method to actually perform any logging.  Subclasses can
     * override this single method to capture all log activity.
     *
     * @param record the LogRecord to be published
     */
    public void log(LogRecord record) {
        if (!isLoggable(record.getLevel())) {
            return;
        }
        Filter theFilter = filter;
        if (theFilter != null && !theFilter.isLoggable(record)) {
            return;
        }

        // Post the LogRecord to all our Handlers, and then to
        // our parents' handlers, all the way up the tree.

        Logger logger = this;
        while (logger != null) {
            final Handler[] loggerHandlers = isSystemLogger
                ? logger.accessCheckedHandlers()
                : logger.getHandlers();

            for (Handler handler : loggerHandlers) {
                handler.publish(record);
            }

            final boolean useParentHdls = isSystemLogger
                ? logger.useParentHandlers
                : logger.getUseParentHandlers();

            if (!useParentHdls) {
                break;
            }

            logger = isSystemLogger ? logger.parent : logger.getParent();
        }
    }
```